src/sw.ts

```diff
@@ -12,29 +12,53 @@ import {
   setDefaultHandler,
 } from "workbox-routing";
 
-export const assets = [...publicDirAssets, ...emittedAssets].filter((asset) => !excludeAssets.includes(asset));
+const serviceWorkerName = self.location.pathname.replace(/^\//, "");
+const useExcludedAssets = [
+  ...excludeAssets,
+  serviceWorkerName,
+  `build/${serviceWorkerName}`,
+];
+
+const buildMap: Map<string, string | null> = appBundles.reduce(
+  (map, bundle) => {
+    const bundleName = bundle[0];
+    if (useExcludedAssets.includes(bundleName)) return map;
+
+    const revision = bundle.length === 3 && bundle[2].length;
+    map.set(bundleName, revision ? null : manifestHash);
+    map.set(`build/${bundleName}`, revision ? null : manifestHash);
+    return map;
+  },
+  new Map<string, string | null>(),
+);
+// console.log("buildMap", buildMap);
+export const assets = [...publicDirAssets, ...emittedAssets].filter(
+  (asset) => !useExcludedAssets.includes(asset),
+);
+
 export { routes };
 
+// TODO: check what happens with urls when using a custom base in Vite/Qwik
+const urlsToEntriesMatcher = /^build\/q-/;
 function urlsToEntries(urls: string[], hash: string): PrecacheEntry[] {
-  const matcher = /^build\/q-/;
   return urls.map((url) => {
-    const match = url.match(matcher);
     // use null revision, removing the revision or using undefined will cause workbox warnings in runtime
-    return match ? { url, revision: null } : { url, revision: hash };
+    return buildMap.has(url)
```

Comment:
We should provide sw precaching manifest at build time, we don't need this logic at runtime.

We also need to check  why `nr build + nr serve` or `nr build-prompt + nr serve` are different from `nr preview` or `nr preview-prompt` (`build command or serve is about SSG (??)?)